---

> 🤖 Daily AI Assistant — **correction + dependency (2026-06-03)**

**The go.mod bump alone does NOT fix the scan** — and currently makes it *worse*.
Live root-cause from this PR's failing scan
([run 26863554119](https://github.com/devantler-tech/ksail/actions/runs/26863554119)):

```
go: go.mod requires go >= 1.26.4 (running go 1.26.3; GOTOOLCHAIN=local)
##[error]govulncheck failed to run (exit 1) - operational error, not a scan result.
```

The `🛡️ Vulnerability Scan` runs `setup-go` with `go-version: stable` (resolves to
**1.26.3**) *and* `go-version-file: go.mod`; setup-go gives `go-version` precedence,
so the scan **ignores go.mod** and always runs under `stable`. Bumping go.mod to
require 1.26.4 therefore turns the scan into an operational error (the runner's local
1.26.3 can't satisfy go.mod and won't auto-upgrade).

The build/test/lint/tidy jobs all **pass** here (they use `go-version-file: go.mod`
*without* a competing `go-version`, and auto-install 1.26.4) — only the vuln scan is
broken by the precedence bug.

**Fix for the scan lives in the shared library:**
[reusable-workflows#284](https://github.com/devantler-tech/reusable-workflows/pull/284)
makes the vuln scan honour `go-version-file` (passes `go-version-input: ''`). The org
required-workflow is pinned to rw `main`, so once #284 lands this PR's next scan runs
under go1.26.4 and clears `GO-2026-5037/5038/5039`.

**Order of operations:** merge rw#284 first, then re-run this PR's vuln scan (or rebase).
This PR (go.mod `1.26.3→1.26.4`) remains correct and necessary — it's what gets the
fixed stdlib into ksail's actual builds and lets the (now go.mod-honouring) scan clear
the advisories. The `System Test (VCluster)` failure is a separate, re-runnable flake.

--- a/.github/workflows/ci.yaml
+++ b/.github/workflows/ci.yaml
@@ -285,7 +285,7 @@ jobs:
   ci-go:
     name: ✅ Validate Go Project
     if: github.event_name == 'push' && github.ref == 'refs/heads/main'
-    uses: devantler-tech/reusable-workflows/.github/workflows/validate-go-project.yaml@8d882e6868d7d7207a708ec1f5ff33b1c90a1770 # v5.4.0
+    uses: devantler-tech/reusable-workflows/.github/workflows/validate-go-project.yaml@bfb04968a21911d080d082e54cef52e0216cffef # v5.4.6
     permissions:
       contents: write
       issues: write

--- a/.golangci.yml
+++ b/.golangci.yml
@@ -2,7 +2,7 @@ version: "2"
 run:
   allow-parallel-runners: true
   modules-download-mode: readonly
-  go: "1.26.3"
+  go: "1.26.4"
 formatters:
   enable:
     - gci

